### DISCUSSION TITLE
Better which-key support

### DISCUSSION THREAD

**Discussion by justinbarclay** [bbatsov/projectile#1818] [Ideas]
@bbatsov, I recently started playing with which-key and noticed that some of Projectile's mappings aren't labelled.

![image](https://user-images.githubusercontent.com/6868388/213820857-a968694c-e78c-4a8c-93ff-1861dbf250c9.png)

I'm able to improve the labelling of nested maps using a which-key helper, `which-key-add-keymap-based-replacements`,
for example: 
```
    (defvar projectile-other-window-map
      (let ((map (make-sparse-keymap)))
        (define-key map (kbd "a") #'projectile-find-other-file-other-window)
        (define-key map (kbd "b") #'projectile-switch-to-buffer-other-window)
        (define-key map (kbd "C-o") #'projectile-display-buffer)
        (define-key map (kbd "d") #'projectile-find-dir-other-window)
        (define-key map (kbd "D") #'projectile-dired-other-window)
        (define-key map (kbd "f") #'projectile-find-file-other-window)
        (define-key map (kbd "g") #'projectile-find-file-dwim-other-window)
        (define-key map (kbd "t") #'projectile-find-implementation-or-test-other-window)
        map))
    ...
    (which-key-add-keymap-based-replacements projectile-command-map
      "4" (cons "other-window" projectile-other-window-map)
      "5" (cons "other-frame" projectile-other-frame-map)
      "s" (cons "search" projectile-search-map))
```
Ending up with:
![image](https://user-images.githubusercontent.com/6868388/213821464-fe23f062-cc4f-4e12-b761-c735c9c6f098.png)

However, would you be receptive of a PR that helps improve the labelling with which-key?

P.S. Thanks for all of your work on Projectile. It's an amazing tool ❤️

**Comment by bbatsov** [comment]
@justinbarclay Yeah, PRs in this direction would be welcome!
